content/_code-samples/markers-and-pagination/js/pagination-with-markers.js

```diff
@@ -0,0 +1,42 @@
+const xrpl = require("xrpl")
+
+async function main() {
+
+  // Create a client and connect to the network.
+  const client = new xrpl.Client("wss://xrplcluster.com/")
+  await client.connect()
+
+  // Specify a ledger to query for info.
+  let ledger = await client.request({
+    "command": "ledger_data",
+    "ledger_index": 500000,
+  })
+
+  // Create a function to run on each API call.
+  function code(){
+    console.log(ledger["result"])
+  }
+ 
+  // Execute function at least once before checking for markers.
+  do {
+    code()
+    
+    if (ledger["result"]["marker"] == null) {
```

Comment:
As a best practice, avoid `==` in JavaScript which performs implicit data type conversion. I think this code will work as written, but if you switch to `===` then it needs to be `undefined` rather than `null` (since `null` is an actual JSON value that can be specified explicitly, but the API does not do that, it just omits the field).

```suggestion
    if (ledger["result"]["marker"] === undefined) {
```